src/backend/marsha/bbb/tests/api/classroom/recordings/test_create_vod.py

```diff
@@ -127,6 +145,38 @@ def test_api_classroom_recording_create_vod_instructor_or_admin(self):
             recording.vod.get_source_s3_key(stamp=to_timestamp(now)),
         )
 
+    def test_api_classroom_recording_create_vod_instructor_or_admin_unknown_recording(
+        self,
+    ):
+        """Instructors and admins should not be able to convert an unknown recording to a VOD."""
+        recording = ClassroomRecordingFactory(
+            started_at="2019-08-21T15:00:02Z",
+        )
+        jwt_token = InstructorOrAdminLtiTokenFactory(resource=recording.classroom)
+
+        self.assertEqual(Video.objects.count(), 0)
+
+        now = timezone.now()
+
+        with mock.patch.object(
+            timezone, "now", return_value=now
+        ), self.assertNumQueries(1):
+            response = self.client.post(
+                f"/api/classrooms/{recording.classroom.id}"
+                f"/recordings/{recording.classroom.id}/create-vod/",
+                HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+                data={"title": "My title"},
+            )
+
+        self.assertEqual(Video.objects.count(), 0)
+        self.assertEqual(response.status_code, 404)
+
+        recording.refresh_from_db()
```

Comment:
I don't see the point to refresh it and not use it later.

Reply:
copy-pastis !